### PR TITLE
Reject invalid boolean input values

### LINKED ISSUE #1157: Reject invalid values for boolean inputs
**Description:**
Boolean inputs are parsed by comparing their uppercased value with `TRUE`. Any other non-empty value is silently interpreted as `false`, so mistakes such as `verify-signature: ture` disable the requested behavior instead of failing clearly. Replace this behavior with strict parsing that accepts only supported boolean values and reports the input name and invalid value.

Apply the behavior consistently to `check-latest`, `force-download`, `set-default`, `verify-signature`, `overwrite-settings`, `show-download-progress`, and `problem-matcher`. Add unit tests for valid values, defaults, casing, whitespace, and invalid values.

**Task version:**
v6

**Platform:**
- [x] Ubuntu
- [x] macOS
- [x] Windows

**Runner type:**
- [x] Hosted
- [x] Self-hosted

**Repro steps:**  
Configure an input with an invalid boolean value, for example:

```yaml
- uses: actions/setup-java@v6
  with:
    distribution: temurin
    java-version: '21'
    verify-signature: ture
```

**Expected behavior:**
The action fails before installation with a descriptive validation error identifying `verify-signature` and its invalid value.

**Actual behavior:**
The invalid value is silently interpreted as `false`, and installation continues without signature verification.

### PR DESCRIPTION
**Pull request by brunoborges**
**Description:**
Boolean input typos currently fall through as `false`, which can silently disable requested behavior. This change makes the shared parser accept only case-insensitive `true` or `false`, preserves existing defaults and whitespace handling, and reports the input name and invalid value for unsupported values.

The stricter behavior applies consistently to all seven boolean inputs that use the helper. The checked-in setup and cleanup bundles are rebuilt to match the TypeScript source.

**Related issue:**
Fixes: #1157

**Check list:**
- [x] Ran `npm run check` locally (format, lint, build, test) and all checks pass.
- [ ] Mark if documentation changes are required.
- [x] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

File: src/util.ts [hunk 1]
```diff
@@ -20,8 +20,21 @@ export function getTempDir() {
...
 
 export function getBooleanInput(inputName: string, defaultValue = false) {
-  return (
-    (core.getInput(inputName) || String(defaultValue)).toUpperCase() === 'TRUE'
+  const inputValue = core.getInput(inputName);
+  const normalizedValue = inputValue.trim().toLowerCase();
+
+  if (!normalizedValue) {
+    return defaultValue;
+  }
```
[the remark this answers is not in the document]
Comment (brunoborges) [reply src/util.ts:28]: Thanks for flagging this. I am leaving the implementation unchanged because JavaScript default parameters apply when an explicit `undefined` is passed, so `defaultValue` becomes `false`; the focused util test, including `[undefined, false]`, passes.
